operator/v1/types_network.go

```diff
@@ -562,6 +562,12 @@ type IPsecConfig struct {
 	// +kubebuilder:validation:Enum=Disabled;External;Full
 	// +optional
 	Mode IPsecMode `json:"mode,omitempty"`
+	// encapsulation uses rfc3949 encapsulation of IKE and IPsec encryption packets
+	// This is applicable only when mode is set to Full.
+	// Default is false.
+	// +optional
+	// +kubebuilder:default:=false
+	Encapsulation bool `json:"encapsulation,omitempty"`
```

Comment:
This sounds like we are creating fields that only apply to certain modes, so, this struct should evolve to become a discriminated union.

It can't be a true union because mode is optional, but we should move any "full" related fields into a struct called `Full` to group them together

We can then use our standard union validation to ensure that the field `full` is specified only when `mode: Full`.

Given this is added after the mode field, that will be an optional union member